### PR TITLE
Add tutorial to explain naming of PyGMT figure elements

### LINKED ISSUE #2026: Explain the naming of figure elements in a PyGMT figure
The figure below explains the anatomy of a matplotlib figure.
image

As you can see, the naming of figure elements is different between matplotlib and PyGMT. 

1. _tick labels_ in matplotlib are called _annotations_ in PyGMT
2. _axes_ in matplotlib is a region for plotting data. I don't think there is an equivalent in PyGMT
3. _axis_ (e.g., x-axis and y-axis)in matplotlib is called _axes_ in PyGMT, and is specified by **WSENwsen**.
4. _spines_ in matplotlib are lines connecting the axis tick marks and noting the boundaries of the data area. In GMT/PyGMT, they're plotted by specifying **l**, **r**, **t**, or **b**.

These differences may confuse users coming from matplotlib. I think it would be good if we can create a similar figure and short tutorial, explaining the naming of figure elements in GMT/PyGMT.

### PR DESCRIPTION
**Pull request by JingHuiTong**
**Description of proposed changes**
Add a tutorial to explain naming of figure elements in a PyGMT
image

**Preview**: https://pygmt-dev--2383.org.readthedocs.build/en/2383/get_started/03_figure_element.html

Fixes #2026

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/tutorials/basics/fig_element.py
Comment (seisman) [inline examples/tutorials/basics/fig_element.py:1]: I feel this tutorial should be put into the `examples/get_started` directory.

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:4]:
```suggestion
The figure shows the naming of figure elements in PyGMT. 
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:6]:
```suggestion
* ``frame``: setting map boundary (**WSNE** or **wsne**), annonate and grid (**afg**), axis label (**+l**) and title (**+t**) in :func:`pygmt.Figure.basemap`
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:3]:
```suggestion
Figure elements
===============
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:8]:
```suggestion
* ``frame``: setting map boundary (**WSNE** or **wsne**), annonate and grid (**afg**), axis label (**+l**) and title (**+t**) in :meth:`pygmt.Figure.basemap`
* :meth:`pygmt.Figure.plot`: plotting the lines or symbols based on ``pen`` or ``style`` parameters, respectively
* :meth:`pygmt.Figure.legend`: showing the naming of lines or symbols while the ``label`` given in :meth:`pygmt.Figure.plot`
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:85]:
```suggestion

# ============Legend
fig.legend()
fig.text(x=7.5, y=17, text="Legend", font="12p,2,darkblue", justify="TL", no_clip=True)
fig.text(x=7.5, y=15.8, text="fig.legend()", font="10p,8", justify="TL", no_clip=True)
```

**Comment by willschlitzer** [comment]
@JingHuiTong Do you get any changes when you run `make format` and `make check`? That should help you with the failing Style Checks.

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:18]:
```suggestion
- :meth:`pygmt.Figure.legend`: showing the naming of lines or symbols while
  the ``label`` is given in :meth:`pygmt.Figure.plot`
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:16]:
```suggestion
- :meth:`pygmt.Figure.text`: plotting text strings whereby the ``font``
  parameter adjusts fontsize, fontstyle, and color.
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:7]: Users can add different plots or / and maps to one figure object.
```suggestion
- :meth:`pygmt.Figure`: having a number of plotting methods. Every new figure
  must start with the creation of a :meth:`pygmt.Figure()` instance
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:12]: Beside `pygmt.Figure.basemap` also other methods have the ``frame`` parameter.
```suggestion
- `
- ``frame``: setting plot boundaries (**WSNE**, **wsne**, or **lbtr**),
   adding annotations, ticks, gridlines (**afg**), axis labels (**+l**),
   and title (**+t**), e.g., in :meth:`pygmt.Figure.basemap`. Detailed
   examples can be found at
   `frame and axes attributes<https://www.pygmt.org/latest/tutorials/basics/frames.html>`_.
```

File: examples/get_started/03_figure_element.py
Comment (weiji14) [inline examples/get_started/03_figure_element.py:54]: Typo. Should it be Annotate or Annotation?
```suggestion
fig.text(x=10, y=-1.4, text="Annotation", **mainexplain)
fig.text(x=10, y=-2.6, text='frame="xa2"', **minorexplain)
# ============y-majorticks
fig.plot(x=-0.2, y=20, style="c1c", pen="2p,darkblue", no_clip=True)
fig.text(x=0, y=23.5, text="Annotation", **mainexplain)
```

File: examples/get_started/03_figure_element.py
Comment (weiji14) [inline examples/get_started/03_figure_element.py:12]: Use relative intersphinx link - https://docs.readthedocs.io/en/stable/guides/intersphinx.html#using-intersphinx
```suggestion
  Detailed examples can be found at :doc:`frame and axes attributes
  </tutorials/basics/frames>`.
```

File: examples/get_started/03_figure_element.py
Comment (weiji14) [inline examples/get_started/03_figure_element.py:74]: Explain the triangle plot (style) and red line (pen) separately?

```suggestion
# ============fig.plot (style)
fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
fig.text(x=6, y=6.5, text="fig.plot(style)", **minorexplain)
# ============fig.plot (pen)
fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
fig.text(x=3, y=4.5, text="fig.plot(pen)", **minorexplain)
```

**Review by weiji14 — COMMENTED** [review]
Hi @JingHuiTong, sorry for the slow response. This example is looking pretty good already, and we're hoping to maybe get it in for the PyGMT v0.10.0 release due next week (1 Sep) if possible. Was there anything else you wanted to add to the example, or can we start to finalize things?

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:4]:
```suggestion
The figure below shows the naming of figure elements in PyGMT.
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:13]:
```suggestion
- :meth:`pygmt.Figure.plot`: plotting lines or symbols based on ``pen``
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:16]:
```suggestion
  parameter adjusts fontsize, fontstyle, and color
```
To be consistent with the the other points.

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:19]:
```suggestion
- :meth:`pygmt.Figure.show`: previewing the content added to the current figure instance
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:9]: Hm, I am not completely happy with the formulation of this bullet point. Actually, the user can pass a combination of the single letters of `WSENE`, `wsne`, and `lbtr`.

**Comment by JingHuiTong** [comment]
> 

Sorry for the slow response, I think the example almost done. I will try to review the comment.

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:9]:
```suggestion
- ``frame``: setting plot or map boundaries (a combination of the single 
  letters of **WSNE**, **wsne**, or **lbtr**), adding annotations, ticks, 
```

**Review by weiji14 — APPROVED** [review]
Thanks @JingHuiTong! We'll merge this after a second approval, and get it in for PyGMT v0.10.0 :smile:

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:80]:
```suggestion
fig.text(x=8, y=16.9, text="Legend", **mainexplain)
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:52]:
```suggestion
fig.text(x=10, y=-1.6, text="Annotation", **mainexplain)
fig.text(x=10, y=-2.8, text='frame="xa2"', **minorexplain)
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:56]:
```suggestion
fig.text(x=0, y=23.4, text="Annotation", **mainexplain)
fig.text(x=0, y=22.2, text='frame="ya5"', **minorexplain)
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:42]:
```suggestion
fig.text(x=12, y=20.8, text="pygmt.Figure()", **minorexplain)
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:71]:
```suggestion
fig.text(x=11.5, y=6.8, text='frame="WStr"', **minorexplain)
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:40]:
```suggestion
# ============ Figure
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:43]:
```suggestion
# ============ Title
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:45]:
```suggestion
# ============ xlabel
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:47]:
```suggestion
# ============ ylabel
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:49]:
```suggestion
# ============ x-majorticks
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:53]:
```suggestion
# ============ y-majorticks
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:57]:
```suggestion
# ============ x-minorticks
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:61]:
```suggestion
# ============ y-minorticks
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:64]:
```suggestion
# ============ Grid
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:68]:
```suggestion
# ============ Map Boundaries
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:72]:
```suggestion
# ============ fig.plot (style)
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:75]:
```suggestion
# ============ fig.plot (pen)
```

File: examples/get_started/03_figure_element.py
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:78]:
```suggestion
# ============ Legend
```

**Review by michaelgrund — COMMENTED** [review]
Some small adjustments to have an almost consistent style regarding the distances between different text boxes.

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:8]: Just remove the trailing white spaces to make the Code Style checks happy 🙂.
```suggestion
- ``frame``: setting plot or map boundaries (a combination of the single
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:9]:
```suggestion
  letters of **WSNE**, **wsne**, or **lbtr**), adding annotations, ticks,
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:10]:
```suggestion
  gridlines (**afg**), axis labels (**+l**), and title (**+t**), e.g.,
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:11]:
```suggestion
  in :meth:`pygmt.Figure.basemap`. Detailed examples can be found at
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:19]:
```suggestion
- :meth:`pygmt.Figure.show`: previewing the content added to the current
```

**Review by michaelgrund — APPROVED** [review]

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:67]: At the marked position (a crossing point of x and y gridlines) it is strictly speaking:
```suggestion
fig.text(x=2, y=18, text='frame="[xg2, yg5]"', **minorexplain)
```

File: examples/get_started/03_figure_element.py
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:77]: Small adjustment of the position of the "fig.plot(style)" label:
```suggestion
# ============ fig.plot (style)
fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
fig.text(x=7, y=6.5, text="fig.plot(style)", **minorexplain)
# ============ fig.plot (pen)
fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
fig.text(x=3, y=4.5, text="fig.plot(pen)", **minorexplain)
```
| currently  | suggestion |
| -------- | ------- |
|fig_plot_style_label_position_old | fig_plot_style_label_position_new |

**Review by yvonnefroehlich — COMMENTED** [review]
Thanks @JingHuiTong for working on this tutorial :slightly_smiling_face:. It will definitely help users coming from other python plotting libraries or MATLAB.

**Comment by weiji14** [comment]
I'm gonna go ahead and apply Yvonne's suggestions and merge this PR for PyGMT v0.10.0.

**Review by seisman — APPROVED** [review]

**Pull request merged**
